src/use-swr-infinite.ts

```diff
@@ -117,10 +118,11 @@ function useSWRInfinite<Data = any, Error = any>(
 
   // keep the data inside a ref
   const dataRef = useRef<Data[]>()
+  const lastFetchedDataRef = useRef<Data[]>()
 
   // actual swr of all pages
   const swr = useSWR<Data[], Error>(
-    firstPageKey ? ['many', firstPageKey] : null,
+    firstPageKey ? ['many', firstPageKey, pageCountRef.current] : null,
```

Comment:
I don't think we should change the key here. It's OK to update both lists when clicking "new page" in the sandbox you shared, since the data (including all pages) is bound to the key.

Reply:
@shuding Thank you for your feedback!

> It's OK to update both lists when clicking "new page"

I misunderstood the expected behavior. In that case, should SWR manage the page count in the cache rather than a ref in the component? Because the ref is stored in each component, so it seems to cause inconsistency between components.

```
A1 -> pageCount: 1, data: 1-1, 1-2, 1-3
A2 -> pageCount: 1, data: 1-1, 1-2, 1-3

A1 changes pageCount to 2

A1 -> pageCount: 2, data: 1-1, 1-2, 1-3, 2-1, 2-2, 2-3
A2 -> pageCount: 1, data: 1-1, 1-2, 1-3, 2-1, 2-2, 2-3 # pageCount should be 2

```

Reply:
Yeah I think so! The page count is current kept in the cache:

```js
pageCountCacheKey = 'len@' + firstPageKey
```

But we also used a ref to handle it too, which brings the inconsistency I believe. Maybe we should only use the cached page count to manage that, and trigger a revalidation when it changes.

Reply:
To support `persistSize: true`, we might have to keep something as a ref to persist the size between different keys.
https://github.com/vercel/swr/pull/490

Reply:
Ah that's true, good catch! 👍 It makes things a little bit complicated here although.

Reply:
I've addressed this.